core/vm/contracts_ec.go

```diff
@@ -73,6 +83,61 @@ func RunECPrecompiledContract(evm *EVM, precompile ECPrecompiledContract, input
 	return output, suppliedGas, err
 }
 
+// bw6761PlonkProofVerifyPrecompile implements the verification of a given PlonK proof. It is assumed that the proof
+// was generated using BW6-761 curve.
+//
+// The input is assumed to encode, in the following order:
+// - a PlonK proof of a circuit,
+// - a verifying key for the proof,
+// - a public witness suitable for verifying the proof.
+//
+// All the above components must be generated for the same circuit definition, over the BW6-761 finite field.
```

Comment:
```suggestion
// All the above components must be generated for the same circuit definition, over the BW6-761 finite field.
//
// Gnark version used by the proving service MUST BE the same as the one used by Quorum. The binary representation
// of the serialized objects is Gnark's internal implementation detail. No assumptions can be made about the byte-level
// structure of the input buffer. The only way of making sure the buffer is structured correctly is to maintain the
// order of the objects and keep Gnark versions synchronized among the interfacing projects.
```

@iamrecursion that's regarding your previous comment